docs/docs/internal-development/create-admin-components-with-theme-support.md

```diff
@@ -0,0 +1,212 @@
+
+```ts
+const theme = createCometTheme({
+    components: {
+        CometAdminMyComponent: {
+            defaultProps: {
+                shadow: true,
+            },
+        },
+    },
+});
+```
+
+### Overriding the styles of a component
+
+E.g.: in every instance of `MyComponent`, the root should have a red border, and the header should be lime.
+
+```ts
+const theme = createCometTheme({
+    components: {
+        CometAdminMyComponent: {
+            styleOverrides: {
+                root: {
+                    borderColor: "red",
+                },
+                header: {
+                    backgroundColor: "lime",
+                },
+            },
+        },
+    },
+});
+```
+
+### Overriding the styles of a single instance of a component
+
+E.g.: in this instance of `MyComponent`, the root should have a red border, and the header should be lime.
+
+```tsx
+<MyComponent
+    sx={{ borderColor: "red" }}
+    slotProps={{ header: { sx: { backgroundColor: "lime" } } }}
+/>
+```
+
+## Example component
+
+```tsx
+import { CometColor } from "@comet/admin-icons";
+import { ComponentsOverrides, Typography } from "@mui/material";
+import { css, styled, SxProps, Theme, useThemeProps } from "@mui/material/styles";
+import React from "react";
+
+import { SlotProps } from "./helpers/SlotProps";
+
+/**
+ * The `{ComponentName}ClassKey` type defines all the individual elements (or slots) of a component or a modifier or alternative style of a component.
+ * - `root` is generally the outermost element of the component.
+ * - `header` in this case would be the header element of the component.
+ * - `hasShadow` would be a modifier class added to the root element when the `shadow` prop is set to `true`.
+ */
+export type MyComponentClassKey = "root" | "header" | "typography" | "icon" | "children" | "hasShadow";
+
+type RootProps = Pick<MyComponentProps, "shadow">;
+
+/**
+ * Each element or sub-component of a Comet Admin component should be created as a "slot".
+ * A slot is created by using the `styled` function from `@mui/material/styles` and passing in the HTML element or component you want to be the base of your slot.
+ * The slot's options require:
+ * - `name`: The name of the component, prefixed by `CometAdmin`.
+ *    This name is used to reference the component in the theme, so it must be the same for every slot in the component.
+ * - `slot`: The name of the slot. This name is used to reference the slot when overriding component styles or default-props.
+ * - `overridesResolver`: The function that applies the styles from the theme's styleOverrides to the slot.
+ *    This should always return the styles of the slot's name, e.g. `styles.root` for the Root slot.
+ *    If the slot has modifier styles, e.g. `styles.hasShadow`, then those should be returned as well when the modifier prop is set to `true`.
+ */
+const Root = styled("div", {
+    name: "CometAdminMyComponent",
+    slot: "Root",
```

Comment:
Technically it doesn't matter but since it's always accessed with lowercase (e.g. in the theme) it makes sense to also have it lowercase here. I used [MUI's implementations](https://github.com/mui/material-ui/blob/v5.14.18/packages/mui-material/src/Button/Button.js#L64) as a guide, it was capitalized there 😄